src/js/utils/background.js

```diff
@@ -138,14 +138,39 @@ export const backgroundStyle = (backgroundArg, theme, textColorArg) => {
   );
 
   if (background.image) {
-    // allow both background color and image, in case the image doesn't fill
-    return css`
+    const backgroundStyles = `
+      ${backgroundColor ? `background-color: ${backgroundColor};` : ''}
       background-image: ${background.image};
       background-repeat: ${background.repeat || 'no-repeat'};
       background-position: ${background.position || 'center center'};
       background-size: ${background.size || 'cover'};
-      ${backgroundColor ? `background-color: ${backgroundColor};` : ''}
+    `;
+
+    // allow both background color and image, in case the image doesn't fill
+    // when image and opacity are used together, we need to use pseudo :before
+    // to ensure that only image and background color are affected by opacity
+    // but not the container contents
+    return css`
       ${textColor ? `color: ${textColor};` : ''}
+      ${!background.opacity
```

Comment:
Can we test the scenario of having background-image, background-color, and background-opacity? And make sure the opacity applies to both the image and the color.

Reply:
Yes, I will test it.

Reply:
I checked this and it applies to both the image and background color : )